app/classifier/index.cjsx

```diff
@@ -431,20 +433,35 @@ Classifier = React.createClass
     taskDescription = @props.workflow.tasks[taskKey]
     annotation = tasks[taskDescription.type].getDefaultAnnotation taskDescription, @props.workflow, tasks
     annotation.task = taskKey
+
+    if @props.workflow.configuration.persist_annotations
+      cachedAnnotation = @props.cacheClassification.isAnnotationCached(taskKey)
+      if cachedAnnotation?
+        annotation = cachedAnnotation
+
     classification.annotations.push annotation
     classification.update 'annotations'
 
   # Back up:
   destroyCurrentAnnotation: ->
+    lastAnnotation = @props.classification.annotations[@props.classification.annotations.length - 1]
+
     @props.classification.annotations.pop()
```

Comment:
You're right, but doing this instead seems to be breaking returning to the first task in a branching workflow. Not sure why yet.